### PR TITLE
Go crash reporting.

### PR DESCRIPTION
**Pull request by ben-clayton**

### CODE REVIEW

**Comment by ben-clayton** [comment]
Note - I haven't seen a test report appear, so its entirely possible I've screwed up the multipart message somehow. Please let me know if the message looks screwy to you.

File: core/app/crash/reporting/encoder.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+type encoder struct {
+	appName    string
+	appVersion string
+	osName     string
+	osVersion  string
+	stacktrace string
+}
+
+const multipartBoundary = "::multipart-boundary::"
```
Comment (baldwinn860) [inline core/app/crash/reporting/encoder.go]: Is there a reason we set this manually? Typically a UUID is desired I think.

**Comment by baldwinn860** [comment]
I see the test report appear on the staging server now.

File: core/app/crash/reporting/encoder.go (hunk 1 above)
Comment (ben-clayton) [reply core/app/crash/reporting/encoder.go]: I made it fixed for testing. I figured the chances of `"::multipart-boundary::"` being used in the stack trace or other data was slim to none, so altering the boundary string on tests was fairly pointless.

**Pull request merged**